noir_stdlib/src/lib.nr

```diff
@@ -18,5 +18,19 @@ mod compat;
 #[builtin(println)]
 fn println<T>(_input : T) {}
 
+#[oracle(println)]
+unconstrained fn println_oracle<T, A>(_input: T) -> Field {}
+
+#[oracle(println_format)]
+unconstrained fn println_format_oracle<T, A, N>(_msg: T, _args: [A; N]) -> Field {}
+
+unconstrained fn println_new<T>(input: T) {
+    println_oracle(input);
+}
+
+unconstrained fn println_format<T, A, N>(msg: T, args: [A; N]) {
```

Comment:
This is currently restrictive as it requires the arguments of the format function to all be of the same type. I will make a PR based off of this one that proposes a method for enabling many different args inside of the format builtin

Reply:
This is being worked upon here: https://github.com/noir-lang/noir/pull/1952